### ISSUE TITLE
ADC pin numbering not as documented?

### ISSUE THREAD

**Issue by DeeJay** [MrYsLab/telemetrix-rpi-pico#9]
ADC pin numbering not as documented?

```
>>>
>>> board = telemetrix_rpi_pico.TelemetrixRpiPico()
TelemetrixRpiPico:  Version 1.1

Copyright (c) 2020-2021 Alan Yorinks All Rights Reserved.

Opening all potential serial ports...
        /dev/ttyACM0
Serial compatible device found and connected to /dev/ttyACM0
Retrieving pico ID...
Pico Unique ID: [230, 97, 100, 7, 227, 90, 0, 0]

Retrieving Telemetrix4pico firmware ID...
Telemetrix4pico firmware version: 1.1
>>> board.get_pico_pins()
{0: 255, 1: 255, 2: 255, 3: 255, 4: 255, 5: 255, 6: 255, 7: 255, 8: 255, 9: 255, 10: 255, 11: 255, 12: 255, 13: 255, 14: 255, 15: 255, 16: 255, 17: 255, 18: 255, 19: 255, 20: 255, 21: 255, 22: 255, 25: 255, 26: 255, 27: 255, 28: 255}
>>> board.set_pin_mode_analog_input(1)
>>> board.set_pin_mode_analog_input(2)
>>> board.set_pin_mode_analog_input(3)
>>> board.set_pin_mode_analog_input(4)
>>> board.set_pin_mode_analog_input(0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.9/dist-packages/telemetrix_rpi_pico/telemetrix_rpi_pico.py", line 711, in set_pin_mode_analog_input
    raise RuntimeError('Invalid ADC Number')
RuntimeError: Invalid ADC Number
>>> board.set_pin_mode_analog_input(5)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.9/dist-packages/telemetrix_rpi_pico/telemetrix_rpi_pico.py", line 711, in set_pin_mode_analog_input
    raise RuntimeError('Invalid ADC Number')
RuntimeError: Invalid ADC Number
>>> board.get_pico_pins()
{0: 255, 1: 255, 2: 255, 3: 255, 4: 255, 5: 255, 6: 255, 7: 255, 8: 255, 9: 255, 10: 255, 11: 255, 12: 255, 13: 255, 14: 255, 15: 255, 16: 255, 17: 255, 18: 255, 19: 255, 20: 255, 21: 255, 22: 255, 25: 255, 26: 255, 27: 5, 28: 5, 29: 5, 30: 5}
>>> board.shutdown()
>>>

```
Valid ADC pin numbers appear to be [1,2,3,4], not 0-3 as documented?

get_pico_pins() subsequently reports  [spurious?] gp29 and gp30?

**Comment by MrYsLab** [comment]
@DeeJay 
Thank you for reporting this. You are correct, ADC 0 was not accepted as a valid value. I will correct this.

For get_pico_pins the report is correct with one exception, which also will be fixed. 
The exception is that GPIO 25 is not a valid GPIO pin.

The reason you see pins 26-30 being reported is that ADC0 (pin 26), ADC1 (pin 27), and ADC2 (pin.28) are mapped to actual physical pins. There are two internal ADC pins. ADC3 is the ADC reference voltage which I treat as pin 29, and ADC4 is the processor temperature, which I treat as pin 30. 

The reason pins 29 and 30 are not reported in the first call to get_pico_pins is because neither ADC3 nor ADC4 have been initialized yet. Once they, they appear as pin 29 and 30 in the output.

To sum up, I will be making two changes. The first is to correct the exception you are seeing. The second is to remove GPIO 25 from the valid list of pins.

Please let me know if you agree with two fixes. Once you let me know, I will publish an updated release. Also, does get_pico_pins make sense now?

**Comment by DeeJay** [comment]
Thank you for your prompt response.

More complicated than I thought: I imagined a simple 'off by one' error.

Your proposals seem correct, but perhaps in addition the docs should explain why '4' is a valid argument?

**Comment by MrYsLab** [comment]
Ok. I will create a release and notify you here when it is available on pypi. I will also add some notes to the User's Guide.

**Comment by MrYsLab** [comment]
Release 1.2 is now available. I also updated the documentation in the code and in the [user's guide.](https://mryslab.github.io/telemetrix-rpi-pico/analog_input/)
